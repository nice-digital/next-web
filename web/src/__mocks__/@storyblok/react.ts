export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
