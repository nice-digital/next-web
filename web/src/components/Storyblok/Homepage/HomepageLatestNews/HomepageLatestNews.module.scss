@use '@nice-digital/nds-core/scss/spacing';
@use '@nice-digital/nds-core/scss/utils';
@use '~/styles/spacing' as *;

.latestNews {
  margin-bottom: $homepage-large-spacing;
}

.title {
  margin: 0 0 utils.rem(spacing.$x-large);
}
