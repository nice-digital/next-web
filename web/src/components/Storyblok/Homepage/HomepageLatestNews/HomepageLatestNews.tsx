import { StoryblokStory } from "storyblok-generate-ts";

import { FeaturedStory } from "@/components/Storyblok/News/FeaturedStory/FeaturedStory";
import { NewsGrid } from "@/components/Storyblok/News/NewsGrid/NewsGrid";
import { type NewsStory } from "@/types/News";

import styles from "./HomepageLatestNews.module.scss";

interface LatestNewsProps {
	featuredStory?: StoryblokStory<NewsStory>;
	latestNews: StoryblokStory<NewsStory>[];
}

export const HomepageLatestNews: React.FC<LatestNewsProps> = ({
	featuredStory,
	latestNews,
}: LatestNewsProps) => {
	return (
		<section className={styles.latestNews} data-tracking="latest-news">
			<h2 className={styles.title}>Latest from NICE</h2>
			{featuredStory && <FeaturedStory story={featuredStory} />}
			<NewsGrid news={latestNews} />
		</section>
	);
};
