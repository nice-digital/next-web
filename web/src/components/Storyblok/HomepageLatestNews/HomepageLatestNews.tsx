import { type StoryblokStory } from "storyblok-generate-ts";

import {
	type BlogPostStoryblok,
	type NewsArticleStoryblok,
} from "@/types/storyblok";

interface LatestNewsProps {
	featuredStory?:
		| string
		| StoryblokStory<BlogPostStoryblok>
		| StoryblokStory<NewsArticleStoryblok>;
	latestNews: (
		| StoryblokStory<BlogPostStoryblok>
		| StoryblokStory<NewsArticleStoryblok>
	)[];
}

export const HomepageLatestNews: React.FC<LatestNewsProps> = ({
	featuredStory,
	latestNews,
}: LatestNewsProps) => {
	console.log("Latest news in the component:", latestNews);

	return (
		<>
			{featuredStory && "Got a featured story!"}
			<p>Latest news has {latestNews.length} items!</p>
		</>
	);
};
