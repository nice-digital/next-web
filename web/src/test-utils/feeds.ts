import fs from "fs/promises";
import path from "path";

import MockAdapter from "axios-mock-adapter";

import { client } from "../feeds";

const mockDataBaseDirectory = path.resolve(
		__dirname,
		"../",
		"__mocks__",
		"__data__"
	),
	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
	publicationsBaseUrl = new URL(process.env.SERVER_FEEDS_PUBLICATIONS_ORIGIN!),
	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
	inDevBaseUrl = new URL(process.env.SERVER_FEEDS_INDEV_ORIGIN!);

export const axiosJSONMock = new MockAdapter(client, {
	onNoMatch: "throwException",
});

/**
 * Adds default JSON endpoint mocks for publications and indev feeds.
 *
 * Maps the incoming request URL to static JSON files on the file system in the mock data folder
 */
export const addDefaultJSONFeedMocks = (): MockAdapter =>
	axiosJSONMock.onGet().reply(async (config) => {
		if (!config.url) throw Error(`URL was empty in Axios mock`);

		const { host, pathname } = new URL(config.url),
			isHTML = config.headers?.Accept === "text/html";

		let filePath = mockDataBaseDirectory;

		if (host === publicationsBaseUrl.host) {
			filePath = path.join(filePath, "publications");
		} else if (host === inDevBaseUrl.host) {
			filePath = path.join(filePath, "inDev");
		} else
			throw Error(
				`Unknown URL host for mocking. Expected either ${publicationsBaseUrl.host} or ${inDevBaseUrl.host} but got ${host}`
			);

		filePath = path.join(
			filePath,
			pathname.toLowerCase() + (isHTML ? ".html" : ".json")
		);

		try {
			await fs.access(filePath);
		} catch {
			throw Error(`Could not find mock JSON file with path ${filePath}`);
		}

		const fileContents = await fs.readFile(filePath, "utf8");

		return [
			200,
			isHTML ? fileContents : !fileContents ? "" : JSON.parse(fileContents),
		];
	});
