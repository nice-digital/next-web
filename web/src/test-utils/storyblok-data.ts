// Dump any useful test objects in here for mocking Storyblok responses

import { type StoryblokStory } from "storyblok-generate-ts";

import {
	type AuthorStoryblok,
	type RichtextStoryblok,
	type BlogPostStoryblok,
	type NewsArticleStoryblok,
	AssetStoryblok,
	PodcastStoryblok,
	InDepthArticleStoryblok,
} from "@/types/storyblok";

export const mockStoryblokStory = {
	id: 123456789,
	uuid: "abc123",
	name: "Title of the story",
	created_at: "1980-08-27T11:12:13.194Z",
	published_at: "2024-01-31T15:16:15.194Z",
	first_published_at: "2024-01-30T15:16:15.194Z",
	slug: "i-am-a-slug",
	full_slug: "i/am/a-very-full-slug",
	sort_by_date: null,
	position: 3,
	tag_list: ["rhubarb", "custard"],
	is_startpage: false,
	parent_id: 987654321,
	meta_data: null,
	group_id: "5625b207-7bb8-4410-aac3-3d7d5743f97a",
	release_id: null,
	lang: "default",
	path: undefined,
	alternates: [],
	default_full_slug: "default-full-slug",
	translated_slugs: [
		{
			path: "/translated-slug-path",
			name: "Translated slug name",
			lang: "en",
		},
	],
};

export const mockAuthorBlok: StoryblokStory<AuthorStoryblok> = {
	...mockStoryblokStory,
	name: "Test Author title",
	content: {
		component: "author",
		name: "Bob Holness",
		jobTitle: "Blockbusters host",
		image: {
			id: 123,
			filename:
				"http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29",
			name: "Bob",
			alt: "Bob the host",
		},
		_uid: "123",
	},
};

export const mockRichText: RichtextStoryblok = {
	type: "doc",
	content: [
		{
			type: "heading",
			attrs: {
				level: 2,
			},
			content: [
				{
					text: "A mock h2 heading in a rich text field",
					type: "text",
				},
			],
		},
		{
			type: "heading",
			attrs: {
				level: 3,
			},
			content: [
				{
					text: "A mock h3 heading",
					type: "text",
				},
			],
		},
		{
			type: "heading",
			attrs: {
				level: 4,
			},
			content: [
				{
					text: "A mock h4 heading",
					type: "text",
				},
			],
		},
		{
			type: "heading",
			attrs: {
				level: 5,
			},
			content: [
				{
					text: "A mock h5 heading",
					type: "text",
				},
			],
		},
		{
			type: "heading",
			attrs: {
				level: 6,
			},
			content: [
				{
					text: "A mock h6 heading",
					type: "text",
				},
			],
		},
		{
			type: "paragraph",
			content: [
				{
					type: "text",
					text: "Example text content.",
				},
			],
		},
		{
			type: "paragraph",
			content: [
				{
					type: "image",
					attrs: {
						id: "13894387",
						alt: "Kitten wearing mittens",
						src: "http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29",
						title: "",
						source: "",
						copyright: "",
						meta_data: {},
					},
				},
			],
		},
		{
			type: "blockquote",
			content: [
				{
					type: "paragraph",
					content: [
						{
							text: "A inline quote in the rich text field of a news article. ",
							type: "text",
						},
					],
				},
			],
		},
	],
};

export const mockNewsArticle: StoryblokStory<NewsArticleStoryblok> = {
	...mockStoryblokStory,
	id: 987654,
	name: "Test news article title",
	content: {
		title: "Test news article title",
		date: "2024-01-31T15:16:15.19",
		introText: "Test intro text",
		image: {
			id: 123,
			filename:
				"http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29",
			name: "A kitten being a kitten",
			alt: "Kitten mittens",
		},
		_uid: "abc123",
		component: "newsArticle",
		content: { ...mockRichText },
		resources: [
			{
				_uid: "8f019cf9-3743-4126-94cb-092898b43636",
				link: {
					id: "e9c6fedf-5d63-4468-b29f-8884d5300ad8",
					url: "",
					linktype: "story",
					fieldtype: "multilink",
					cached_url: "test-new-nc-2",
				},
				title: "Test related link 1",
				component: "relatedLink",
				_editable:
					'<!--#storyblok#{"name": "relatedLink", "space": "271255", "uid": "8f019cf9-3743-4126-94cb-092898b43636", "id": "435600482"}-->',
			},
		],
		relatedNews: [
			{
				_uid: "50c7601a-1f1f-48f1-96e9-4f16f688fd7a",
				date: "2024-01-31 00:00",
				link: {
					id: "",
					url: "https://youtube.com/@niceorguk",
					linktype: "url",
					fieldtype: "multilink",
					cached_url: "https://youtube.com/@niceorguk",
				},
				title: "Test related news link 1",
				component: "relatedNewsLink",
				publisher: "NICE",
				_editable:
					'<!--#storyblok#{"name": "relatedNewsLink", "space": "271255", "uid": "50c7601a-1f1f-48f1-96e9-4f16f688fd7a", "id": "435600482"}-->',
			},
		],
	},
};

export const mockBlogPost: StoryblokStory<BlogPostStoryblok> = {
	...mockStoryblokStory,
	name: "Test blog post title",
	content: {
		title: "Test blog post title",
		date: "2024-02-20",
		introText: "Let's play Blockbusters!",
		author: [{ ...mockAuthorBlok }],
		image: {
			id: 456,
			filename:
				"http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29",
			name: "Cat cat cat",
			alt: "A kitty cat",
		},
		_uid: "def456",
		component: "blogPost",
		content: { ...mockRichText },
	},
};

export const mockPodcastPage: StoryblokStory<PodcastStoryblok> = {
	...mockStoryblokStory,
	name: "Test podcast title",
	content: {
		_uid: "2663bc46-cefe-4878-8c86-ecc78cc1f65e",
		date: "2024-04-17 12:00",
		image: {
			id: 15267557,
			alt: "",
			name: "",
			focus: "",
			title: "",
			source: "",
			filename:
				"https://local-host-test-nice-org.com/f/243782/413x297/6a7748fdb4/stroke.PNG",
			copyright: "",
			fieldtype: "asset",
			meta_data: {},
			is_private: false,
			is_external_url: false,
		},
		component: "podcast",
		introText:
			"It is estimated that there are around 85,000 strokes every year in England, with over 1 million people currently living with its long-term effects.\n" +
			"\n" +
			"We catch up with Jon Brown, chair of the independent committee and Professor Stephen Hill, lay member on the committee about NICE’s updated guidance on stroke rehabilitation.",
		description: {
			type: "doc",
			content: [
				{
					type: "paragraph",
					content: [
						{
							text: "It is estimated that there are around 85,000 strokes every year in England, with over 1 million people currently living with its long-term effects.",
							type: "text",
						},
					],
				},
			],
		},
		soundcloudEmbedID: "1648857267",
	},
};

export const mockImageAsset: AssetStoryblok = {
	filename:
		"http://local-host-test-nice-org.com/f/243782/420x234/7a576133e7/register-for-board-meeting.PNG/m/867x0/filters:format%28jpeg%29:quality%2880%29",
	id: 654654654,
	name: "Delicious bacon",
};

export const mockIndepthArticle: StoryblokStory<InDepthArticleStoryblok> = {
	...mockStoryblokStory,
	name: "sample indepth test",
	id: 476215653,
	uuid: "7bb356cb",
	content: {
		_uid: "cd984458-0e14-4fd6-871d-469e9cd3a845",
		date: "2024-04-10 12:00",
		link: {
			id: "",
			url: "https://local-host-test-nice-org.com/news/blogs/lessons-learned-from-the-multi-agency-advisory-service",
			linktype: "url",
			fieldtype: "multilink",
			cached_url:
				"https://local-host-test-nice-org.com/news/blogs/lessons-learned-from-the-multi-agency-advisory-service",
		},
		image: mockImageAsset,
		title: "sample test",
		component: "inDepthArticle",
		introText:
			"Clíodhna Ní Ghuidhir and Rebecca Boffa outline the common challenges that developers of artificial intelligence (AI) tech face when navigating their way to the health and care market, and link to their series of articles, which provide further advice on what can be done to avoid these.",
	},
};
