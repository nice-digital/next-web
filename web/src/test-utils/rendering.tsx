/* eslint-disable import/export, import/no-unresolved -- We're deliberately overrind the default render from @testing-library with a custom render method */
import {
	render as testingLibrRender,
	RenderResult,
} from "@testing-library/react";
import { HeadManagerContext } from "next/dist/shared/lib/head-manager-context.shared-runtime";
import { RouterContext } from "next/dist/shared/lib/router-context.shared-runtime";
import { NextRouter } from "next/router";
import { ReactElement, useEffect } from "react";
import { renderToString } from "react-dom/server";

import NextWebApp from "../pages/_app";

import type { AppProps } from "next/app";

/**
 *
 */
export const mockRouter: NextRouter = {
	basePath: "",
	pathname: "/",
	route: "/",
	asPath: "/",
	query: {},
	push: jest.fn(),
	replace: jest.fn(),
	reload: jest.fn(),
	back: jest.fn(),
	forward: jest.fn(),
	prefetch: jest.fn(),
	beforePopState: jest.fn(),
	events: {
		on: jest.fn(),
		off: jest.fn(),
		emit: jest.fn(),
	},
	isFallback: false,
	isLocaleDomain: false,
	isPreview: false,
	isReady: false,
};

/**
 * Convenience method for getting a mock router with the given properties.
 *
 * @example
 * 	getMockRouter({ pathname: "/about" })
 *
 * @param routerParams Optional router properties that get merged with the default mock router
 * @returns A new mock router object
 */
export const getMockRouter = (
	routerParams?: Partial<NextRouter>
): NextRouter => ({
	...mockRouter,
	...routerParams,
});

/**
 * Collect tags from next/head: https://edibleco.de/3Anu92S
 */
const HeadProvider: React.FC = ({ children }) => {
	let head: JSX.Element[];

	useEffect(() => {
		global.document.head.insertAdjacentHTML(
			"afterbegin",
			renderToString(<>{head}</>) || ""
		);
	});

	return (
		<HeadManagerContext.Provider
			value={{
				updateHead: (state) => {
					head = state;
				},
				mountedInstances: new Set(),
			}}
		>
			{children}
		</HeadManagerContext.Provider>
	);
};

export type GetWrapperOptions = {
	router: NextRouter;
	wrapper: RenderOptions["wrapper"];
};
const getAppWrapper =
	({ router, wrapper: Wrapper }: GetWrapperOptions) =>
	({ children }: { children: ReactElement }) =>
		(
			<HeadProvider>
				<RouterContext.Provider value={router}>
					<NextWebApp
						Component={() => (
							<>{Wrapper ? <Wrapper>{children}</Wrapper> : children}</>
						)}
						pageProps={{}}
						router={router as AppProps["router"]}
					/>
				</RouterContext.Provider>
			</HeadProvider>
		);

export type RenderOptions = Parameters<typeof testingLibrRender>[1] & {
	router?: Partial<NextRouter>;
};

/**
 * Custom render method the given component, wrapped in `NextWebApp` and a `HeadProvider`.
 *
 * @see https://testing-library.com/docs/react-testing-library/setup/#custom-render
 *
 * @param ui The element to render
 * @param options Optional render options that get passed to underlying React Testing Lib render method
 * @returns The render result
 */
const customRender = (
	ui: ReactElement,
	{ wrapper, router, ...options }: RenderOptions = {}
): RenderResult =>
	testingLibrRender(ui, {
		wrapper: getAppWrapper({ router: getMockRouter(router), wrapper }),
		...options,
	});

export * from "@testing-library/react";
export { customRender as render };
