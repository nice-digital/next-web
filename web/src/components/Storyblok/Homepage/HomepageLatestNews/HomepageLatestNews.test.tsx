import { render, screen } from "@testing-library/react";
import { StoryblokStory } from "storyblok-generate-ts";

import { mockNewsArticle, mockBlogPost } from "@/test-utils/storyblok-data";
import type { NewsStory } from "@/types/News";

import { HomepageLatestNews } from "./HomepageLatestNews";

// Mock a featured story
const featuredStory: StoryblokStory<NewsStory> = {
	...mockNewsArticle,
	id: 333333,
	name: "Featured story title",
	content: {
		...mockNewsArticle.content,
		introText: "Featured story intro text",
		image: {
			id: 234,
			filename: "https://placebacon.net/300/210?image=1",
			name: "Delicious bacon",
		},
	},
};

// Mock a second article with a different ID so we avoid duplicate key warnings
const secondNewsArticle = {
	...mockNewsArticle,
	id: 987654321,
};

const latestNews: StoryblokStory<NewsStory>[] = [
	mockNewsArticle,
	mockBlogPost,
	secondNewsArticle,
];

describe("Homepage latest news component", () => {
	it("should match snapshot without a featured story", () => {
		const { container } = render(
			<HomepageLatestNews latestNews={latestNews} />
		);
		expect(container).toMatchSnapshot();
	});

	it("should match snapshot with a featured story", () => {
		const { container } = render(
			<HomepageLatestNews
				latestNews={latestNews}
				featuredStory={featuredStory}
			/>
		);
		expect(container).toMatchSnapshot();
	});

	it("should render a featured story if one is supplied", () => {
		render(
			<HomepageLatestNews
				latestNews={latestNews}
				featuredStory={featuredStory}
			/>
		);

		expect(
			screen.getByRole("heading", { level: 3, name: "Featured story title" })
		).toBeInTheDocument();
	});

	it("should not render a featured story if one is not supplied", () => {
		render(<HomepageLatestNews latestNews={latestNews} />);

		expect(
			screen.queryByRole("heading", { level: 3, name: "Featured story title" })
		).toBeNull();
	});
});
