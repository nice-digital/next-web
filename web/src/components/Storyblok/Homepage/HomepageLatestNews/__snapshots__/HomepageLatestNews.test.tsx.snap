// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage latest news component should match snapshot with a featured story 1`] = `
<div>
  <section
    class="latestNews"
  >
    <h2
      class="title"
    >
      Latest from NICE
    </h2>
    <article
      class="story"
    >
      <div
        class="imageContainer"
        style="background-image: url(https://placebacon.net/300/210?image=1/m/868x0/filters:quality%2880%29);"
      />
      <div
        class="content"
      >
        <h3
          class="heading"
        >
          <a
            href="/i/am/a-very-full-slug"
          >
            Featured story title
          </a>
        </h3>
        <p>
          Featured story intro text
        </p>
        <footer>
          <span
            class="tag tag--outline"
            data-component="tag"
          >
            News
          </span>
          <span
            class="date"
          >
            31 January 2024
          </span>
        </footer>
      </div>
    </article>
    <ol
      class="newsGrid"
    >
      <li
        class="newsItem"
      >
        <article
          class="newsCard "
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/i/am/a-very-full-slug"
            style="background-image: url(https://placekitten.com/200/286/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/i/am/a-very-full-slug"
              >
                Test news article title
              </a>
            </h3>
            <p>
              Test intro text
            </p>
            <footer>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
                <span
                  class="date"
                >
                  31 January 2024
                </span>
              </div>
            </footer>
          </div>
        </article>
      </li>
      <li
        class="newsItem"
      >
        <article
          class="newsCard "
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/i/am/a-very-full-slug"
            style="background-image: url(https://placekitten.com/200/139/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/i/am/a-very-full-slug"
              >
                Test blog post title
              </a>
            </h3>
            <p>
              Let's play Blockbusters!
            </p>
            <footer>
              <div
                class="author"
              >
                <p
                  class="author"
                >
                  by 
                  Bob Holness
                  , 
                  Blockbusters host
                </p>
              </div>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  Blog
                </span>
                <span
                  class="date"
                >
                  20 February 2024
                </span>
              </div>
            </footer>
          </div>
        </article>
      </li>
      <li
        class="newsItem"
      >
        <article
          class="newsCard "
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/i/am/a-very-full-slug"
            style="background-image: url(https://placekitten.com/200/286/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/i/am/a-very-full-slug"
              >
                Test news article title
              </a>
            </h3>
            <p>
              Test intro text
            </p>
            <footer>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
                <span
                  class="date"
                >
                  31 January 2024
                </span>
              </div>
            </footer>
          </div>
        </article>
      </li>
    </ol>
  </section>
</div>
`;

exports[`Homepage latest news component should match snapshot without a featured story 1`] = `
<div>
  <section
    class="latestNews"
  >
    <h2
      class="title"
    >
      Latest from NICE
    </h2>
    <ol
      class="newsGrid"
    >
      <li
        class="newsItem"
      >
        <article
          class="newsCard "
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/i/am/a-very-full-slug"
            style="background-image: url(https://placekitten.com/200/286/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/i/am/a-very-full-slug"
              >
                Test news article title
              </a>
            </h3>
            <p>
              Test intro text
            </p>
            <footer>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
                <span
                  class="date"
                >
                  31 January 2024
                </span>
              </div>
            </footer>
          </div>
        </article>
      </li>
      <li
        class="newsItem"
      >
        <article
          class="newsCard "
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/i/am/a-very-full-slug"
            style="background-image: url(https://placekitten.com/200/139/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/i/am/a-very-full-slug"
              >
                Test blog post title
              </a>
            </h3>
            <p>
              Let's play Blockbusters!
            </p>
            <footer>
              <div
                class="author"
              >
                <p
                  class="author"
                >
                  by 
                  Bob Holness
                  , 
                  Blockbusters host
                </p>
              </div>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  Blog
                </span>
                <span
                  class="date"
                >
                  20 February 2024
                </span>
              </div>
            </footer>
          </div>
        </article>
      </li>
      <li
        class="newsItem"
      >
        <article
          class="newsCard "
        >
          <a
            aria-hidden="true"
            class="imageContainer"
            href="/i/am/a-very-full-slug"
            style="background-image: url(https://placekitten.com/200/286/m/868x0/filters:quality%2880%29);"
            tabindex="-1"
          >
             
          </a>
          <div>
            <h3
              class="heading"
            >
              <a
                href="/i/am/a-very-full-slug"
              >
                Test news article title
              </a>
            </h3>
            <p>
              Test intro text
            </p>
            <footer>
              <div>
                <span
                  class="tag tag--outline"
                  data-component="tag"
                >
                  News
                </span>
                <span
                  class="date"
                >
                  31 January 2024
                </span>
              </div>
            </footer>
          </div>
        </article>
      </li>
    </ol>
  </section>
</div>
`;
